### ISSUE TITLE
Saved event fired on old related model

### ISSUE THREAD

**Issue by TivoSoho** [laravel/framework#42221]
- Laravel Version: 8.x - 9.x

### Description:
This is the source for touching owners. As you can see it fires "saved" event on an old model if it existed beforehand.
```
    /**
     * Touch the owning relations of the model.
     *
     * @return void
     */
    public function touchOwners()
    {
        foreach ($this->getTouchedRelations() as $relation) {
            $this->$relation()->touch();

            if ($this->$relation instanceof self) {
                $this->$relation->fireModelEvent('saved', false);

                $this->$relation->touchOwners();
            } elseif ($this->$relation instanceof Collection) {
                $this->$relation->each->touchOwners();
            }
        }
    }
```

### Steps To Reproduce:
1. Have something set to touch the parent
2. Have that something have that relation loaded
3. Save the child
4. Try to access the new timestamp in parent saved event - it is still the old data

### Correct steps:
If relation exists and it wants to fire a saved event - it should refresh the data. Or there should be some option as whether we want to use the updated timestamp (or any other data we add with touch, for example modifiers) and thus it should refresh the data. To me it is a bug because when you are listening for a saved event you expect to get the SAVED model - not old data.

**Comment by driesvints** [comment]
It seems to me that you want a change in behavior which is a feature request. You're free to attempt a PR if you like. Thanks

**Comment by TivoSoho** [comment]
Can you clarify what then constitutes as a bug? If a relationship is not loaded - it retrieves it fresh and it has the correct data. However if relationship is loaded, the saved event is fired on data that is not only NOT saved, but is missing the actual data that WAS saved. Meaning, if you modify the relationship and it does touch, the touch is done via rawUpdate, but the even is fired on the changed model that doesn't even have the fresh timestamp. But as pointed out - if it was not loaded and it does retrieve it, then everything is correct.

Basically the entire saved event itself IS a bug. It does not behave like the other events (which should have the correct data), but also you cannot really auto-refresh it either in case the loaded relationship was changed. Basically, one should retrieve it fresh and fire the event on that and not on the already loaded related model.
